pkg/runtime/runtime.go

```diff
@@ -191,25 +191,24 @@ func (rt *ResourceGraphDefinitionRuntime) ResourceDescriptor(id string) Resource
 // whether the resource variables are resolved or not, and whether the resource
 // readiness conditions are met or not.
 func (rt *ResourceGraphDefinitionRuntime) GetResource(id string) (*unstructured.Unstructured, ResourceState) {
-	// Did the user set the resource?
-	r, ok := rt.resolvedResources[id]
-	if ok {
+	rt.mu.RLock()
+	defer rt.mu.RUnlock()
+
+	if r, ok := rt.resolvedResources[id]; ok {
 		return r, ResourceStateResolved
 	}
-
-	// If not, can we process the resource?
-	resolved := rt.canProcessResource(id)
-	if resolved {
+	if rt.canProcessResource(id) {
```

Comment:
you have quite a bunch of microchanges in this pr that make it tough to review. in this case for example the lock addition is fine but the other refactor messes up the diff quite a bit. maybe try to make this pr really as atomic as it needs to be as the changeset is huge

Reply:
This particular microchange I've reverted in 11289f3ec5a4ebdb2384e75d8fd92b4844fcd6a6